### ISSUE TITLE
Predictable API responses

### ISSUE THREAD

**Issue by innovate-invent** [galaxyproject/galaxy#10399]
Currently the API returns a mix of possible responses. Many frameworks expect a predictable schema to unmarshal into native data structures. 

`POST /api/tool_shed_repositories/install_repository_revision` can respond with a status response or a list of repository objects depending on server state. This requires inspection of the response json before it can be unpacked into a language native data structure.

`POST /api/histories` returns `{err_msg:"", err_code: ""}` while `PUT /api/histories/<id>` returns `{message: "", code: ""}` on error. 

A weaker example of this is WorkflowInvocation.outputs has a mixture of hda and hdca objects. The exception here is they have a common model_class field that allows distinguishing during unmarshalling.

Ideally the content type header would specify the response schema id as `application/vnd.galaxy.<schema id>+json`.
This requires that API request and response schemas be documented.

https://en.wikipedia.org/wiki/Media_type

**Comment by jmchilton** [comment]
Fair point but can we close this as a duplicate of https://github.com/galaxyproject/galaxy/issues/3211 to some degree? Or can you replace this issue with specific actionable problems new developers could hack on?

**Comment by innovate-invent** [comment]
It looks like #3211 is a dead issue? Hasn't been touched in 3-4 years.
This issue is just proposing cleaning up the existing and inconsistent API responses. Reuse a canned error response template across the codebase.

This task is to review the existing API responses and clean up, #3211 proposes a complete rehash of the API, including a new framework.

**Comment by jmchilton** [comment]
> is a dead issue

It has been hacked on by multiple developers and still remains a long term plan. I also don't particularly consider it a rehash of the whole API since we want to preserve backward compatibility as we go.

> Reuse a canned error response template across the codebase.

I mean - we do this, we have a set of best practices for how to do this and have test cases. I think it would be better to track specific spots we're letting older style errors out.

**Comment by innovate-invent** [comment]
What is the intended error response format?

**Comment by innovate-invent** [comment]
I just found that `DELETE /api/quotas/{id}` returns free form text rather than json when responding
```
400 Bad Request
The server could not comply with the request since
it is either malformed or otherwise incorrect.
        
Quota names must be unique and a quota with that name already exists, so choose another name.
```
